### PR TITLE
feat: Implementation of CsvDataLoader

### PR DESCRIPTION
**Pull request by VicenteVigueras**
Few things:
1. I created a public static void main because I wanted to log and debug stuff in my code, the following command was just not working for me for some reason:
```java
./gradlew run --console=plain --args="--loader=com.codedifferently.lesson12.factory.vicentevigueras"
```
2. At the moment, the code only reads media_items. 
3. The code is having trouble reading item.edition because, for instance, "Morning Edition, March 23, 2024" contains a comma inside of it, and I am splitting at every comma.
4. Lastly, opencsv and commoncsv libraries were helpful.

### CODE REVIEW

**Comment by anthonydmays** [comment]
> Few things:
> 
> 1. I created a public static void main because I wanted to log and debug stuff in my code, the following command was just not working for me for some reason:
> 
> ```java
> ./gradlew run --console=plain --args="--loader=com.codedifferently.lesson12.factory.vicentevigueras"
> ```
> 
> 2. At the moment, the code only reads media_items.
> 3. The code is having trouble reading item.edition because, for instance, "Morning Edition, March 23, 2024" contains a comma inside of it, and I am splitting at every comma.
> 4. Lastly, opencsv and commoncsv libraries were helpful.

The command for you would be:

```java
./gradlew run --console=plain --args="--loader=com.codedifferently.lesson12.factory.vicentevigueras.CsvDataLoader"
```

**Comment by VicenteVigueras** [comment]
Tried implementing method for checkout models. 

Since it broke everything, I decided to keep my code clean and go PR with the stuff that does work. 

The methods that I have so far work with Lists, the missing stuff works with Maps, that's the source of my issue. So I'm thinking that I'll have to make each element contain key value pairs? I am working on it, lots of refactoring.

**Comment by VicenteVigueras** [comment]
**Finishing Lesson12 Assignment**
* - Wed, March 27th at 11:12PM (Still on time with extension given during lecture)*

I know this isn't ideal, but bear with me:
1. getCheckedOutItems() method it gave me a null pointer exception. The hashmap storing all of the data had null elements. Which made sense, because some guests didn't checkout any items. 
2. When the unit test tried to "values().stream().mapToInt(List::size)" it didn't work because when mapToInt receives the values it tries to retrieve the size of the list, and obviously a null does not have a size. Therefore, I had to modify the test case to handle this error. 

I tried my best at explaining my reasoning based on my knowledge and research.

File: lesson_12/io/io_app/src/test/java/com/codedifferently/lesson12/factory/LibraryCsvDataLoaderTest.java
Comment (anthonydmays) [inline lesson_12/io/io_app/src/test/java/com/codedifferently/lesson12/factory/LibraryCsvDataLoaderTest.java:1]: Must undo the change to the test.

**Review by anthonydmays — CHANGES_REQUESTED** [review]

File: lesson_12/io/io_app/src/test/java/com/codedifferently/lesson12/factory/LibraryCsvDataLoaderTest.java
Comment (VicenteVigueras) [reply lesson_12/io/io_app/src/test/java/com/codedifferently/lesson12/factory/LibraryCsvDataLoaderTest.java:1]: Done.

File: lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/vicentevigueras/CsvDataLoader.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    // Create object named model that is an instance of LibraryDataModel
+    var model = new LibraryDataModel();
+    // Creates a list and loads with everything inside of file path
+    model.mediaItems = readMediaItems("csv/media_items.csv");
+    model.guests = readGuests("csv/guests.csv");
+    Map<String, List<CheckoutModel>> checkoutsByGuestEmail =
+        getCheckedOutItems("csv/checked_out_items.csv");
+    for (var guest : model.guests) {
+      var checkouts = checkoutsByGuestEmail.get(guest.email);
+      if (checkouts != null) {
```
[the remark this answers is not in the document]
Comment (VicenteVigueras) [reply lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/vicentevigueras/CsvDataLoader.java]: I did: " if (checkouts == null) {
        checkouts = new ArrayList<>();
      }" which should take care of it.

**Review by anthonydmays — APPROVED** [review]

**Comment by anthonydmays** [comment]
> Hey Anthony, did my code work :')? I had so much work put into this assignment. It looked like it was having an initialization error, I was trying to troubleshoot that.

Yea it's fine. There was an instruction I should have given everyone to prevent this issue, but it's easy enough to fix myself. You're good.

**Pull request merged**